site/multiDim/MultiDimDataPageSettingsPanel.tsx

```diff
@@ -1,118 +1,141 @@
-import { useCallback, useMemo, useRef, useState } from "react"
-import {
-    DimensionEnriched,
-    MultiDimDimensionChoices,
-} from "@ourworldindata/types"
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
 import { faCaretDown } from "@fortawesome/free-solid-svg-icons"
 import cx from "classnames"
-import { MultiDimDataPageConfig, mapValues } from "@ourworldindata/utils"
+import * as _ from "lodash-es"
+import { useMemo, useState } from "react"
+import {
+    Button,
+    Collection,
+    Header,
+    ListBox,
+    ListBoxItem,
+    ListBoxSection,
+    Popover,
+    Select,
+    Text,
+} from "react-aria-components"
 import { OverlayHeader, RadioButton } from "@ourworldindata/components"
-import { useTriggerOnEscape, useTriggerWhenClickOutside } from "../hooks.js"
-
-const DimensionDropdown = (props: {
-    dimension: DimensionEnriched
-    currentChoiceSlug: string
-    onChange?: (slug: string, valueSlug: string) => void
-}) => {
-    const { dimension } = props
-
-    const [active, setActive] = useState(false)
-
-    const toggleVisibility = useCallback(() => setActive(!active), [active])
-    const hide = useCallback(() => setActive(false), [])
-
-    const dropdownRef = useRef<HTMLDivElement>(null)
-
-    useTriggerOnEscape(hide)
-    useTriggerWhenClickOutside(dropdownRef, active, hide)
+import {
+    Choice,
+    DimensionEnriched,
+    MultiDimDimensionChoices,
+} from "@ourworldindata/types"
+import { MultiDimDataPageConfig } from "@ourworldindata/utils"
 
-    const currentChoice = dimension.choicesBySlug[props.currentChoiceSlug]
-    if (currentChoice === undefined)
-        throw new Error(
-            `Unexpected error: Invalid choice slug ${props.currentChoiceSlug} for dimension ${dimension.slug}`
-        )
+function DimensionItem({ choice }: { choice: Choice }) {
+    return (
+        <ListBoxItem
+            className="md-menu__radio-button"
+            id={choice.slug}
+            textValue={choice.name}
+        >
+            {({ isSelected }) => (
+                <RadioButton
+                    checked={isSelected}
+                    onChange={_.noop}
+                    label={
```

Comment:
Should this not rather go into a radio group, and properly implement `onChange`?